### PR TITLE
HTMLRenderer: Strip newline characters

### PR DESCRIPTION
**Pull request by tsipinakis**
Make the html renderer strip newline characters from the text to prevent them from being passed down to stfl and causing rendering errors.

(Perhaps they should be replaced with a space character?)

### CODE REVIEW

**Comment by Minoru** [comment]
> (Perhaps they should be replaced with a space character?)

They definitely should. Newlines, tabs, spaces—all this stuff is usually treated as whitespace by the renderers.

Also, the code will become much shorter if rewritten using `utils::replace_all`.

File: test/htmlrenderer.cpp [hunk 1]
```diff
@@ -555,6 +555,19 @@ TEST_CASE("htmlrenderer: whitespace is erased at the beginning of the paragraph"
 	REQUIRE(links.size() == 0);
 }
 
+TEST_CASE("htmlrenderer: newlines are ignored.") {
```
Comment (Minoru) [inline test/htmlrenderer.cpp]: We don't put period at the end of test description.

**Comment by tsipinakis** [comment]
Oh, didn't see there already was a replace_all function. Updated the commits to use replace_all to replace newlines with space chars.

**Comment by tsipinakis** [comment]
Coveralls doesn't appear to like it when PRs are rebased. Sorry for the spam.

**Comment by Minoru** [comment]
Thanks for your work! No worries about the spam, Coveralls is just doing its job after all ;)

**Pull request merged**